runtime/src/main/scala/com/soundcloud/twinagle/ServerEndpointBuilder.scala

```diff
@@ -7,8 +7,10 @@ import scalapb.{GeneratedMessage, GeneratedMessageCompanion, Message}
 
 class ServerEndpointBuilder(extension: EndpointMetadata => Filter.TypeAgnostic) {
   // TODO: refactor types to aliases?
-  def build[Req <: GeneratedMessage with Message[Req] : GeneratedMessageCompanion, Rep <: GeneratedMessage with Message[Rep] : GeneratedMessageCompanion](rpc: Req => Future[Rep], endpoint: EndpointMetadata): (EndpointMetadata, Service[Request, Response]) = {
+  def build[Req <: GeneratedMessage with Message[Req] : GeneratedMessageCompanion, Resp <: GeneratedMessage with Message[Resp] : GeneratedMessageCompanion](rpc: Req => Future[Resp], endpointMetadata: EndpointMetadata): (EndpointMetadata, Service[Request, Response]) = {
     val httpService: Service[Request, Response] = new ServiceAdapter(rpc)
-    endpoint -> extension(endpoint).toFilter.andThen(httpService)
+    endpointMetadata -> extension(endpointMetadata).toFilter.andThen(
```

Comment:
nit: maybe use the same notation as in the clients now that we've got a bigger stack:
```scala
def build[...] = {
  val httpService: Service[Request, Response] =
    extension(endpointMetadata).toFilter andThen
    new TracingFilter(endpointMetadata, isClient = false) andThen
    new ServiceAdapter(rpc)
  endpointMetadata -> httpService
}